runtime/doc/map.txt

```diff
@@ -1430,6 +1430,145 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview* {nvim-api}
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the preview command flag. The preview
+flag may only be used through the Lua API (see |nvim_create_user_command()|).
+
+The preview callback must be a Lua function. When command preview is
+triggered, the preview callback will be called with 3 arguments, an `opts` table
+which is the same as the table passed to |nvim_create_user_command()| callbacks,
+the preview namespace number that's used to set highlights for the preview,
+and the preview buffer number that's used for the preview window when
+`inccommand=split`. Note that if `inccommand=nosplit`, the preview buffer number
+will not be provided.
```

Comment:
better to write what will be provided instead, `nil` ? Then the reader can directly infer they should write `if preview_buf ~= nil then --[[ do stuff ]] end`

Reply:
done